src/fread.c

```diff
@@ -1841,19 +1843,19 @@ int freadMain(freadMainArgs _args) {
   if (initialBuffRows > INT32_MAX) STOP("Buffer size %lld is too large\n", initialBuffRows);
   nth = imin(nJumps, nth);
 
-
+  if (verbose) DTPRINT("[11] Read the data\n");
   read:  // we'll return here to reread any columns with out-of-sample type exceptions
-  {
-  if (verbose) {
-    DTPRINT("[11] Read the data\n");
-    DTPRINT("  jumps=[%d..%d), chunk_size=%llu, total_size=%llu\n", jump0, nJumps, (llu)chunkBytes, (llu)(lastRowEnd-pos));
-  }
+  if (verbose) DTPRINT("  jumps=[%d..%d), chunk_size=%llu, total_size=%llu\n",
+                       jump0, nJumps, (llu)chunkBytes, (llu)(lastRowEnd-pos));
   ASSERT(allocnrow <= nrowLimit, "allocnrow(%llu) < nrowLimit(%llu)", (llu)allocnrow, (llu)nrowLimit);
   #pragma omp parallel num_threads(nth)
   {
     int me = omp_get_thread_num();
```

Comment:
you can add here
```
bool reportProgress = false;
#pragma omp master
reportProgress = args.showProgress;
```
then everywhere where you check `(me==0 && args.showProgress)` you can just replace with `reportProgress`